File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.js
function solution(num) {
    
    var answer = 0;
    
    while(num !=1 && answer < 500){
        if(num%2 == 0){
            num/=2;
        }else{
            num*=3;
            num++;
        }
        answer++;
    }
    
    if(answer == 500) answer = -1;
    
    return answer;
}

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        
        int answer = 0;
        long number = num;
        
        while(number != 1 && answer < 500){
            if(number%2 == 0){
                number/=2;
            }else{
                number*=3;
                number++;
            }
            answer ++;
        }
        
        if(answer == 500) answer = -1;
        
        return answer;
    }
}